Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -122,11 +122,7 @@ public extension Array where Element: Equatable {
     @discardableResult
     mutating func removeDuplicates() -> [Element] {
         // Thanks to https://github.com/sairamkotha for improving the method
-        self = reduce(into: [Element]()) {
-            if !$0.contains($1) {
-                $0.append($1)
-            }
-        }
+        self = withoutDuplicates()
```

Comment:
I know that this makes sense for the code reuse point of view and there are other extensions on the project that does that, but since we are always learning and trying to improve, making a copy the returning array to `self` maybe is not the best option. Could we implement a more specialized version of this that doesn't make copy the array? Maybe some using `removeAll(where: { check for duplication })` I think this could work.
Same apply to the other method in this PR :))

Reply:
Hello @LucianoPAlmeida 
I think we should leave `func removeDuplicates() -> [Element]` when `Element` is `Equatable` as it is. On the other hand I think we should add a version for `Element: Hashable`.